### ISSUE TITLE
Using WSClient on SSL opens lot of connections to server

### ISSUE THREAD

**Issue by GajendraNaidu** [playframework/playframework#4837]
We are using Play Framework 2.3.4. When WSClient living in Play server makes SSL to another play server, opens many connections. Eventually the targetted Play server CPU utilization goes to 100% and goes into kind unresponsive mode.

Number of requests that are being made to server on average 500/sec. Play Server is on CentOS7. At OS level  we see 20k to 30k socket connections opened with client. This happens only when SSL.enabled.

**Comment by wsargent** [comment]
Do you have a reproducible test case?

**Comment by GajendraNaidu** [comment]
We have two play servers says P1 and P2.

I have below code in P1 some where making HTTP requests to P2. Number of requests for second on an average 500 per second.

  def sendRequest(msgId:String, msgType:String, reqId: String, stringToSend: String, request: WSRequest): Unit = {
    var request = WS.url(statsUrl + msg.uri).withHeaders("Accept" -> "application/json")
    val futureResponse = request.post(stringToSend)
    futureResponse.onComplete {
      case Success(response) => {
        log.info("[{}-{}] Received Http Status {}, Response {}", Array(connectionId, reqId, response.status, response.body))
      }
      case Failure(f) => {
        log.warning("[{}-{}] {}", Array(connectionId, reqId, f.getMessage))
      }
    }
  }

Without SSL number of connections open from P1 to P2 is around 1K.
With SSL number of connections open from P1 to P2 is around 20K to 30K. 

If you need more info please let me know

**Comment by wsargent** [comment]
The only thing that stands out is you have two request objects.

If you see the Play server at 100% or dying because of OOM, try dumping the stack and the heap: http://stackoverflow.com/questions/407612/how-to-get-a-thread-and-heap-dump-of-a-java-process-on-windows-thats-not-runnin

**Comment by GajendraNaidu** [comment]
We analysed dumps, we saw the bad thing is only number of socket connections other things are fine.

Now we migrated to Play Framework version 2.4.2. Now on SSL, CPU Utilization is fine but number of open sockets increase to 200 thousand in a week. Server is working fine but in increasing in number of connections is concern here. we are sending 7MB of data in 348 requests per second. We see increase number of connections to 200 thousand after a week of system running.
